src/custom-style-element.js

```diff
@@ -0,0 +1,134 @@
+/**
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+/*
+Wrapper over <style> elements to co-operate with ShadyCSS
+
+Example:
+<custom-style>
+  <style>
+  ...
+  </style>
+</custom-style>
+*/
+
+'use strict';
+
+let ShadyCSS = window.ShadyCSS;
+
+let enqueued = false;
+
+let alwaysUpdate = false;
+
+let customStyles = [];
+
+let hookFn = null;
+
+let readying = false;
+
+/*
+If a page only has <custom-style> elements, it will flash unstyled content,
+as all the instances will boot asynchronously after page load.
+
+Calling ShadyCSS.updateStyles() will force the work to happen synchronously
+*/
+function enqueueDocumentValidation() {
+  if (enqueued) {
+    return;
+  }
+  enqueued = true;
+  if (window.HTMLImports) {
+    window.HTMLImports.whenReady(validateDocument);
+  } else if (document.readyState === 'complete') {
+   validateDocument();
+  } else if (!readying) {
+    readying = true;
+    document.addEventListener('readystatechange', () => {
+      if (document.readyState === 'complete') {
+        validateDocument();
+      }
+    });
+  }
+}
+
+function validateDocument() {
+  requestAnimationFrame(() => {
+    if (enqueued || alwaysUpdate) {
```

Comment:
Seems like `alwaysUpdate` and the `elementsHaveApplied` flag in the ShadyCSS module are basically the same. Reuse?